packages/server/src/instances/service.ts

```diff
@@ -141,44 +123,52 @@ export class InstanceService extends Service {
     return this.emitter.emit(InstanceEvents.Initialized, conduitId)
   }
 
-  async get(conduitId: uuid): Promise<ConduitInstance<any, any>> {
-    const cached = this.cache.get(conduitId)
-    if (cached) {
-      return cached
-    }
-
+  async start(conduitId: uuid) {
     const conduit = (await this.conduitService.get(conduitId))!
+    const provider = (await this.providerService.getById(conduit.providerId))!
     const channel = this.channelService.getById(conduit.channelId)
-    const instance = channel.createConduit()
+
+    if (channel.has(provider.name)) {
+      return
+    }
 
     try {
       await this.distributedService.using(`lock_dyn_instance_setup::${conduitId}`, async () => {
-        await instance.setup(conduitId, conduit.config, this.app)
+        await channel.start(provider.name, conduit.config)
       })
-      this.cache.set(conduitId, instance, channel.lazy && this.lazyLoadingEnabled ? undefined : Infinity)
-
       await this.emitter.emit(InstanceEvents.Setup, conduitId)
```

Comment:
It would make more sense IMHO if we were to send events with the same name as the function that triggers them! So in this case, shouldn't we use Start instead of setup? Or maybe we should rename the function to setup instead of start? What do you think?